lib/capistrano/ext/multistage.rb

```diff
@@ -22,7 +22,7 @@
   on :load do
     if stages.include?(ARGV.first)
       # Execute the specified stage so that recipes required in stage can contribute to task list
-      find_and_execute_task(ARGV.first) if ARGV.any?{ |option| option =~ /-T|--tasks|-e|--explain/ }
+      find_and_execute_task(ARGV.first) unless (ARGV & %w(-T --tasks -e --explain)).empty?
```

Comment:
I think this would be clearer if you just fixed the regex problem:
```
if ARGV.any? { |option| %w[-T --tasks -e --explain].include?(option) }
```
It might run a little faster too since you don't need the full intersection--you can stop as soon as there's a match, which is what `any?` and `include?` do.
(If you have to use `%w`, you might as well use `%w[]` since it looks array-like.)
Can you send this back as a PR to Captistrano?